apps/emqx_management/src/emqx_mgmt_api_plugins.erl

```diff
@@ -456,8 +456,8 @@ delete_package(Name) ->
 
 %% for RPC plugin update
 ensure_action(Name, start) ->
-    _ = emqx_plugins:ensure_enabled(Name),
     _ = emqx_plugins:ensure_started(Name),
+    _ = emqx_plugins:ensure_enabled(Name),
```

Comment:
The order is important here: first start a plugin then enable it in the conf.
Besides the case when config can be enabled successfully but then fail to be started, there is one more (specific) case when the order is important:

1. Node `A` has plugin `P` which is installed but not started, so it is not listed in the conf (`emx_conf:get([plugins,states])` is empty).
2. Node `B` joins the cluster (node `A`), it doesn't copy plugin `P` because it is not listed in the conf yet (since it's not started on any node).
3. Plugin `P` is started from the dashboard (on any node), and is now being started on all the nodes (via  `emqx_mgmt_api_plugins_proto_v1:ensure_action/2`).
4.  `emqx_plugins:ensure_started(Name)` must now be called before `ensure_enabled` on node `B`, because it does ensure that plugin exists (will copy it from node `A').
5. After this, `emqx_plugins:ensure_enabled/1` can be safely called on node `B`. It will read the plugin description and add it to the `[plugins,states]` conf as enabled.